mempool/lanes_info.go

```diff
@@ -0,0 +1,64 @@
+package mempool
+
+import (
+	"slices"
+
+	"github.com/cometbft/cometbft/types"
+)
+
+type LanesInfo struct {
+	lanes       []types.Lane
+	defaultLane types.Lane
+}
+
+// Query app info to return the required information to initialize lanes.
+func FetchLanesInfo(laneList []uint32, defLane types.Lane) (*LanesInfo, error) {
```

Comment:
I would rename this function to `BuildLanesInfo` or something similar. In the prototype, this function was fetching (querying) the info from the app, but that's not the case anymore.